src/main/java/net/snowflake/client/jdbc/SnowflakeTimeWithSessionTimezone.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright (c) 2012-2020 Snowflake Computing Inc. All rights reserved.
+ */
+
+package net.snowflake.client.jdbc;
+
+import java.sql.Time;
+import java.sql.Timestamp;
+import java.time.LocalDateTime;
+import java.time.ZoneId;
+import java.time.ZoneOffset;
+import java.time.format.DateTimeFormatter;
+import java.util.TimeZone;
+
+/**
+ * Time with toString() overridden to display time values in session timezone. Only relevant for
+ * timestamp objects fetched as times. Normal time objects do not have a timezone associated with
+ * them.
+ */
+public class SnowflakeTimeWithSessionTimezone extends Time {
```

Comment:
Same here. Considered that we can still construct this class without timezone as parameter, I feel like this class name might make some confusion?

Reply:
Changed to SnowflakeTimeWithTimezone instead!